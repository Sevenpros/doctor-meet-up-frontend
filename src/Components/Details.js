import React, { useEffect } from 'react';
import { useParams } from 'react-router';
import { useDispatch, useSelector } from 'react-redux';
import { fetchDoctor } from '../Redux/doctors/doctor';

const Details = () => {
  const doctor = useSelector((state) => state.doctor);
  const { id } = useParams();
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchDoctor(id));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <div className="container d-flex justify-content-center align-items-center">
      <div>
        <img src={doctor.photo} alt="Doctor-Img" className="doctor-img" />
      </div>
      <div className="container d-flex justify-content-center align-items-center">
        <table className="table">
          <thead>
            <tr>
              <th scope="col">First Name</th>
              <th scope="col">Last Name </th>
              <th scope="col">City</th>
              <th scope="col">Contact Number</th>
              <th scope="col">E-mail</th>
              <th scope="col">Bio</th>
              <th scope="col">Specialization</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th scope="row">{doctor.first_name}</th>
              <td>{doctor.last_name}</td>
              <td>{doctor.city}</td>
              <td>{doctor.phone}</td>
              <td>{doctor.email}</td>
              <td>{doctor.bio}</td>
              <td>{doctor.specialization}</td>

            </tr>
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default Details;
