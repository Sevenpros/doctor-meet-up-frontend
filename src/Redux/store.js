import { configureStore } from '@reduxjs/toolkit';
// import logger from 'redux-logger';
import doctorsReducer from './doctors/doctors';
import doctorReducer from './doctors/doctor';

const store = configureStore({
  reducer: {
    doctors: doctorsReducer,
    doctor: doctorReducer,
  },
  // middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});
export default store;
