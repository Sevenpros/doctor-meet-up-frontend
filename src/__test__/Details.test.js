import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

/* eslint-disable-next-line */
import store from '../redux/store';
import Details from '../Components/Details';

describe('Details tests', () => {
  it('render Details test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Details />
        </Router>
      </Provider>,

    );
    expect(tree).toMatchSnapshot();
  });
});
