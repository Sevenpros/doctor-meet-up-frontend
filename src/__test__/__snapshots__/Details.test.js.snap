// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details tests render Details test 1`] = `
<div
  className="container-det"
>
  <div
    className="doctor-div"
  >
    <img
      alt="Doctor-Img"
      className="doc-img"
    />
    <button
      className="back-btn"
      type="button"
    >
      <a
        className="link-light"
        href="/doctors"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 12.796V3.204L4.519 8 10 12.796zm-.659.753-5.48-4.796a1 1 0 0 1 0-1.506l5.48-4.796A1 1 0 0 1 11 3.204v9.592a1 1 0 0 1-1.659.753z"
          />
        </svg>
      </a>
    </button>
  </div>
  <div
    className="doctor-det"
  >
    <h2>
       
    </h2>
    <p>
      Specialization:
       
    </p>
    <table
      className="table"
    >
      <tr>
        <td>
          City
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Contact Number
        </td>
        <td />
      </tr>
      <tr>
        <td>
          E-mail
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Bio
        </td>
        <td />
      </tr>
    </table>
    <button
      className="reserve-btn"
      type="button"
    >
      <a
        className="link-light"
        href="/bookings/new"
        onClick={[Function]}
      >
        Reserve
      </a>
    </button>
  </div>
</div>
`;
